src/Cameras/XR/babylon.webXRManagedOutputCanvas.ts

```diff
@@ -0,0 +1,62 @@
+module BABYLON {
+    /**
+     * Creates a canvas that is added/removed from the webpage when entering/exiting XR
+     */
+    export class WebXRManagedOutputCanvas implements IDisposable {
+        private _canvas: Nullable<HTMLCanvasElement> = null;
+        /**
+         * xrpresent context of the canvas which can be used to display/mirror xr content
+         */
+        public canvasContext: Nullable<WebGLRenderingContext> = null;
+        /**
+         * Initializes the canvas to be added/removed upon entering/exiting xr
+         * @param helper the xr expreince helper used to trigger adding/removing of the canvas
+         * @param canvas The canvas to be added/removed (If not specified a full screen canvas will be created)
+         */
+        public constructor(helper: WebXRExperienceHelper, canvas?: HTMLCanvasElement) {

```

Comment:
I'm wondering why you need to provide a canvas? Could it be jsut an engine? or could we have a canvas which is different from the main engine one?

Reply:
This is the canvas that the xr content is displayed on. Initially I wanted it to just use the current canvas that the engine uses but this isn't supported by the current spec (see issue I created https://github.com/immersive-web/webxr/issues/341). So right now I recommend just having a second full screen canvas, which is what this function will create if canvas isn't specified.